sklearn/exceptions.py

```diff
@@ -154,3 +155,12 @@ class UndefinedMetricWarning(UserWarning):
     .. versionchanged:: 0.18
        Moved from sklearn.base.
     """
+
+
+class SklearnDeprecationWarning(DeprecationWarning):
```

Comment:
Deriving SklearnDeprecationWarning from UserWarning would be backwards incompatible - if someone uses warning filters to suppress sklearn deprecation warning they will stop working.